src/dropin.js

```diff
@@ -358,24 +357,37 @@ Dropin.prototype.clearSelectedPaymentMethod = function () {
 };
 
 Dropin.prototype._setUpDataCollector = function () {
-  var config = assign({}, this._merchantConfiguration.dataCollector, {client: this._client});
-
-  this._model.asyncDependencyStarting();
+  var self = this;
+  var config = assign({}, self._merchantConfiguration.dataCollector, {client: self._client});
 
-  dataCollector.create(config).then(function (instance) {
-    this._dataCollectorInstance = instance;
-    this._model.asyncDependencyReady();
-  }.bind(this)).catch(function (err) {
-    this._model.cancelInitialization(new DropinError({
+  global.braintree.dataCollector.create(config).then(function (instance) {
+    self._dataCollectorInstance = instance;
+    self._model.asyncDependencyReady();
+  }).catch(function (err) {
+    self._model.cancelInitialization(new DropinError({
       message: 'Data Collector failed to set up.',
       braintreeWebError: err
     }));
-  }.bind(this));
+  });
+};
+
+Dropin.prototype._loadDataCollectorScript = function (callback) {
+  var script = document.createElement('script');
```

Comment:
Should this logic be extracted into a helper function? I think we do this a similar thing for checkoutjs.

```js
Dropin.prototype._createScriptTag = function (id, src, callback) {
  var script = document.createElement('script');
  
  this._model.asyncDependencyStarting();

  script.src = src;
  script.id = id;
  script.async = true;
  script.addEventListener('load', callback);

  return script;
}
```

```js
var braintreeWebVersion = this._client.getVersion();
var dataCollectorSrc = 'https://js.braintreegateway.com/web/' + braintreeWebVersion + '/js/data-collector.min.js';
var script = this._createScriptTag(constants.DATA_COLLECTOR_SCRIPT_ID, dataCollectorSrc, this._setUpDataCollector.bind(this));

this._dropinWrapper.appendChild(script);
```

```js
var script = this._createScriptTag(PAYPAL_CHECKOUT_SCRIPT_ID, constants.CHECKOUT_JS_SOURCE, this._setUpDependenciesAndViews();.bind(this));

script.setAttribute('data-log-level', this._merchantConfiguration.paypal.logLevel || DEFAULT_CHECKOUTJS_LOG_LEVEL);

this._dropinWrapper.appendChild(script);
```